Cakeshop doc updates

bmcd opened the pull request:
Updates the docs for the latest Cakeshop release.

Cakeshop no longer runs a node on its own, only connecting to already running nodes. So I don't think there should be a Getting Started -> Using Cakeshop section. So I created a Reference/Cakeshop folder with two pages: Overview and Configuration.

I updated the Cakeshop concepts page to reflect the new changes and removed the Cakeshop FAQ page since it was mostly outdated. Instead I link to the reference pages.

Updated the redirect map to link to these new pages as well.

NicolasMassart commented on docs/Reference/Cakeshop/Overview.md:
```diff
@@ -0,0 +1,79 @@
+
+## Configuration
+
+Cakeshop is a Spring Boot application, so you may place an `application.properties` file in the working directory to override any default configuration values. For more info, see the [configuration page](Configuration.md).
```
Better for search engine: use documentation instead of page which is a too generic term
```suggestion
Cakeshop is a Spring Boot application, so you may place an `application.properties` file in the working directory to override any default configuration values.
For more info, see the [configuration documentation](Configuration.md).
```

NicolasMassart commented on docs/Reference/Cakeshop/Overview.md:
```diff
@@ -0,0 +1,79 @@
+
+## Running via GoQuorum Wizard
+
+The easiest way to use Cakeshop is to generate a GoQuorum network with [GoQuorum Wizard](../../HowTo/GetStarted/Wizard/GettingStarted.md) and choose to deploy Cakeshop alongside the network.
```
```suggestion
The recommended way to use Cakeshop is to generate a GoQuorum network with [GoQuorum Wizard](../../HowTo/GetStarted/Wizard/GettingStarted.md) and choose to deploy Cakeshop alongside the network.
```
statements like easy, easiest, simple,... are to avoid

NicolasMassart commented on docs/Reference/Cakeshop/Overview.md:
```diff
@@ -0,0 +1,79 @@
+## Requirements
+
+* Java 11+
+* Nodejs (if the Nodejs binary on your machine isn't called 'node', see [here](Configuration.md#cakeshop-internals))
```
any Node.js minimal version recommendation?

NicolasMassart commented on docs/Reference/Cakeshop/Overview.md:
```diff
@@ -0,0 +1,79 @@
+
+* Download WAR file
+* Run `java -jar cakeshop.war`
+* Navigate to [http://localhost:8080/](http://localhost:8080/)
```
```suggestion
* Navigate to [`http://localhost:8080/`](http://localhost:8080/)
```

NicolasMassart commented on docs/Reference/Cakeshop/Overview.md:
````diff
@@ -0,0 +1,79 @@
+docker run -p 8080:8080 quorumengineering/cakeshop
+```
+
+Then access the UI at [http://localhost:8080/](http://localhost:8080/)
````
```suggestion
Then access the UI at [`http://localhost:8080/`](http://localhost:8080/)
```

NicolasMassart commented on docs/Reference/Cakeshop/Overview.md:
````diff
@@ -0,0 +1,79 @@
+docker run -p 8080:8080 -v "$PWD/data":/opt/cakeshop/data quorumengineering/cakeshop
+```
+
+An example providing an initial nodes.json in the data directory and configuring it to be used:
````
```suggestion
An example providing an initial `nodes.json` in the data directory and configuring it to be used:
```

NicolasMassart commented on docs/Reference/Cakeshop/Overview.md:
```diff
@@ -0,0 +1,79 @@
+1. Elimination of Cakeshop's managed node in favor of only attaching to existing nodes.
+1. Simplified DB configuration by using Spring Data.
+
+To ensure easy transition, Cakeshop will still look in the locations where v0.11.0 commonly stored the config file. But (1) allows you to now place your config in the folder where you run Cakeshop, or specify a different location using standard spring boot flags, for easier customization.
```
maybe adding these details as sub items in the previous list. It's to prevent the need to go back and forth. Having a list sub item makes it clear that it's detail about the parent item.

NicolasMassart commented on docs/Reference/Cakeshop/Overview.md:
```diff
@@ -0,0 +1,79 @@
+
+(4) means that you will need to update the db-related config values to use spring data. The old `cakeshop.database`, `cakeshop.hibernate`, and `cakeshop.jdbc` settings should change to use `spring.data`, `spring.jpa`, etc. See the [database configuration section](Configuration.md#database) for more info.
+
+Note: Due to an bug that happens when you update from Hibernate 4 to 5, when auto updating the database it will try to recreate some column constraints that don't need to be recreated. These will fail and print a stack trace in the logs because they already exist. No negative effects result from this error, so the best thing to do is to run with `spring.jpa.hibernate.ddl-auto=update` once and the ...
```
Use the `!!!note` admonition (be careful about the 4 spaces increment to make the text part of the note):
```suggestion
!!!note
    Due to a bug that happens when you update from Hibernate 4 to 5, when auto updating the database it will try to recreate some column constraints that don't need to be recreated.
    These will fail and print a stack trace in the logs because they already exist.
    No negative effects result from this error, so the best thing to do is to run with `spring.jpa.hibernate.ddl-auto=update` once and then change `update` to `none` on subsequent runs.
    In production, we do not recommend to use auto update to migrate your database at all, but instead run migrations on the database yourself.
```
As MS guidelines have to be used for all our docs, you can refer to this about the use of "recommend" https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/r/recommend

NicolasMassart commented on docs/Reference/Cakeshop/Overview.md:
```diff
@@ -0,0 +1,79 @@
+
+## Configuration
+
+For more info about how to configure Cakeshop, see the [configuration page](Configuration.md).
```
```suggestion
For more info about how to configure Cakeshop, see the [configuration documentation](Configuration.md).
```
Better for search engine

NicolasMassart reviewed (commented):
Hi, here are some doc writing guidelines comments.

I also added some required changes on the redirects. But first, thanks for having thought about the redirects by yourself!

Once fixed, it will be good for me to approve.

bmcd commented on mkdocs.yml:
```diff
@@ -269,7 +270,7 @@ plugins:
         Permissioning/Enhanced Permissions Model/Permissioning apis.md: Reference/APIs/PermissioningAPIs.md
         Getting Started/api.md: Reference/APIs/PrivacyAPI.md
-        Cakeshop/Cakeshop FAQ.md: Reference/CakeshopFAQ.md
+        Cakeshop/Cakeshop FAQ.md: Reference/Cakeshop/Overview.md
```
fixed, also changed Overview.md to GettingStarted.md to match the content

bmcd commented on mkdocs.yml:
```diff
@@ -254,7 +255,7 @@ plugins:
         Quorum Features/dns.md: HowTo/Configure/dns.md
         PluggableArchitecture/PluginDevelopment.md: HowTo/DevelopPlugins.md
-        Cakeshop/Getting started.md: HowTo/GetStarted/Cakeshop.md
+        Cakeshop/Getting started.md: Reference/Cakeshop/Overview.md
```
fixed

bmcd commented:
I addressed all the feedback above.

I also changed the Cakeshop Overview page to Getting Started instead, since that matches the content better.

I updated that Cakeshop Concepts page to include some new features.

NicolasMassart reviewed (approved):
Looks good to me.
